src/main/java/com/github/cameltooling/lsp/internal/parser/ParserFileHelperUtil.java

```diff
@@ -36,5 +36,27 @@ public String getLine(String text, int line) {
 		}
 		return null;
 	}
+
+	public String getTextUntilPosition(TextDocumentItem document, Position position) {
+		String newLine = System.getProperty("line.separator");
```

Comment:
usually Client implementation of the LSP has their own policy for end of lines. Users are defining the type of end of line in either, thier IDE settings, thei rproject config, their git config. There is no need to take of care of the line separator from the system.